# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.h
# Commit: 8c7b8e63e417
# Full Hash: 8c7b8e63e417be23932362bc9356f2bbf83f8455
# Author: stransky <stransky@redhat.com>
# Date: 2021-12-20 21:51:27
# Regressor Bug: 1746423
# File Overlap Count: 1
# Description:
#   Bug 1746423 [Linux] Remove static XWindow setup and get it on demand, r=rmader
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134024
# ==============================================================================

diff -r aacc456d8cd4 -r 8c7b8e63e417 widget/gtk/nsWindow.h
--- a/widget/gtk/nsWindow.h	Mon Dec 20 10:20:23 2021 +0000
+++ b/widget/gtk/nsWindow.h	Mon Dec 20 10:22:19 2021 +0000
@@ -502,6 +502,9 @@
   void DestroyChildWindows();
   GtkWidget* GetToplevelWidget();
   nsWindow* GetContainerWindow();
+  Window GetX11Window();
+  bool GetShapedState();
+  void EnsureGdkWindow();
   void SetUrgencyHint(GtkWidget* top_window, bool state);
   void SetDefaultIcon(void);
   void SetWindowDecoration(nsBorderStyle aStyle);
@@ -880,11 +883,6 @@
                 GTK_WIDGET_COMPOSIDED_ENABLED = 2} WindowComposeRequest;
   void SetCompositorHint(WindowComposeRequest aState);
   bool ConfigureX11GLVisual();
-
-  Window mXWindow;
-  Visual* mXVisual;
-  int mXDepth;
-  bool mIsShaped;
 #endif
 #ifdef MOZ_WAYLAND
   RefPtr<mozilla::gfx::VsyncSource> mWaylandVsyncSource;
